charts/cluster-addons/templates/openstack/csi-manila.yaml

```diff
@@ -0,0 +1,194 @@
+          csi.storage.k8s.io/node-publish-secret-name: csi-manila
+          csi.storage.k8s.io/node-publish-secret-namespace: {{ .Values.openstack.targetNamespace }}
+        {{- with .Values.openstack.csiManila.cephfsStorageClass }}
+          {{- with .manilaAvailabilityZone }}
+          availability: {{ . }}
+          {{- end }}
+          type: {{ .type }}
+        reclaimPolicy: {{ .reclaimPolicy }}
+        allowVolumeExpansion: {{ .allowVolumeExpansion }}
+        volumeBindingMode: WaitForFirstConsumer
+        {{- with .allowedTopologies }}
+        allowedTopologies: {{ toYaml . | nindent 6 }}
+        {{- end }}
+        {{- end }}
+{{- end }}
+{{- if .Values.openstack.csiManila.nfsStorageClass.enabled }}
+---
+apiVersion: addons.stackhpc.com/v1alpha1
+kind: Manifests
+metadata:
+  name: {{ include "cluster-addons.componentName" (list . "csi-manila-nfs-storageclass") }}
+  labels: {{ include "cluster-addons.componentLabels" (list . "csi-manila") | nindent 4 }}
+  annotations:
+    # Tell Argo to ignore the non-controller owner references for this object
+    argocd.argoproj.io/sync-options: "ControllerReferencesOnly=true"
+spec:
+  clusterName: {{ include "cluster-addons.clusterName" . }}
+  bootstrap: true
+  targetNamespace: {{ .Values.openstack.targetNamespace }}
+  releaseName: csi-manila-nfs-storageclass
+  manifestSources:
+    - template: |
+        {{- with .Values.openstack.csiManila.storageClass }}
+        apiVersion: storage.k8s.io/v1
+        kind: StorageClass
+        metadata:
+          name: {{ .name }}
+          {{- if .isDefault }}
+          annotations:
+            storageclass.kubernetes.io/is-default-class: "true"
+          {{- end }}
+        provisioner: nfs.manila.csi.openstack.org
+        parameters:
+        {{- end }}
+          csi.storage.k8s.io/provisioner-secret-name: csi-manila
+          csi.storage.k8s.io/provisioner-secret-namespace: {{ .Values.openstack.targetNamespace }}
+          csi.storage.k8s.io/controller-expand-secret-name: csi-manila
+          csi.storage.k8s.io/controller-expand-secret-namespace: {{ .Values.openstack.targetNamespace }}
+          csi.storage.k8s.io/node-stage-secret-name: csi-manila
+          csi.storage.k8s.io/node-stage-secret-namespace: {{ .Values.openstack.targetNamespace }}
+          csi.storage.k8s.io/node-publish-secret-name: csi-manila
+          csi.storage.k8s.io/node-publish-secret-namespace: {{ .Values.openstack.targetNamespace }}
+        {{- with .Values.openstack.csiManila.storageClass }}
+          {{- with .manilaAvailabilityZone }}
+          availability: {{ . }}
+          {{- end }}
+          type: {{ .type }}
+        reclaimPolicy: {{ .reclaimPolicy }}
+        allowVolumeExpansion: {{ .allowVolumeExpansion }}
+        volumeBindingMode: WaitForFirstConsumer
+        {{- with .allowedTopologies }}
+        allowedTopologies: {{ toYaml . | nindent 6 }}
+        {{- end }}
+        {{- end }}
```

Comment:
Let's use one `Manifests` resource with multiple `manifestSources` to produce the secret + storage classes.